File: js/agent.js
import * as THREE from 'three';

export class Agent {
    constructor(id, x, y, z, vx, vy, vz, gx, gy, gz, tx, ty, tz, radius, maxSpeed, maxForce, horizon, k) {
        this.id = id;
        this.position = new THREE.Vector3(x, y, z);
        this.velocity = new THREE.Vector3(vx, vy, vz);
        this.goal = new THREE.Vector3(gx, gy, gz);
        this.target = new THREE.Vector3(tx, ty, tz);
        this.radius = radius;
        this.maxSpeed = maxSpeed;
        this.maxForce = maxForce;
        this.horizon = horizon;
        this.k = k;
        this.userData = {};
        this.sphere = new THREE.Sphere(new THREE.Vector3(), radius);
    }

    setData(label, data) {
        this.userData[label] = data;
    }

    getData(label) {
        return this.userData[label]
    }
}

File: py/agent.py
from pygame.math import Vector2

class Agent:
    def __init__(self, id, position, radius, max_speed, max_force, horizon, k, avoid, sidestep):
        self.id = id
        self.position = position
        self.velocity = Vector2(0, 0)
        self.goal = Vector2(position.x, position.y)
        self.target = Vector2(position.x, position.y)
        self.radius = radius
        self.max_speed = max_speed
        self.max_force = max_force
        self.horizon = horizon
        self.k = k
        self.avoid = avoid
        self.sidestep = sidestep